hledger/test/balance/budget.test

```diff
@@ -747,8 +745,7 @@ Budget performance in 2023-08:
 ==============++=================
  income       || 0 [0% of $-200] 
    employment || 0 [0% of $-100] 
-   gifts      || 0               
-     cash     || 0 [0% of $-100] 
+   gifts:cash || 0 [0% of $-100] 
```

Comment:
This looks like a consistency fix, squashing these like a normal balance report would. I think I tried and failed to fix this recently. 👍🏻